### DISCUSSION TITLE
How is it possible to perform multivariate classification tasks?

### DISCUSSION THREAD

**Discussion by lorebon** [wildboar-foundation/wildboar#27] [Q&A] [answered]
Hi everyone,

how is it possible to classify multivariate time-series? For example, when training data is a multidimensional ndarray?

Thanks in advance.

**Comment by isaksamsten** [comment, marked as answer]
Hi!

Many algorithms support multivariate time series out of the box (e.g., `ShapeletForestClassifier`). Just fit it with an `ndarray` `x` of shape `(n_samples, n_dims, n_timestep)` and `y` of shape `(n_samples, )`. Unfortunately, wildboar does not support datasets where each dim have different `n_timestep`. In the example folder there is an example with multivariate time series.

**Comment by lorebon** [comment]
Hi!

 

Thanks for your precious answer, I have a dataset with time-series of equal lengths, and it works well.

 

I have another question: I am interested in counterfactuals, but I cannot fit algorithm “ShapeletForestCounterfactual” to a multivariate time-series.

I try to fit a ndarray of shape (n_dims, n_timesteps) related to one sample, and the corresponding label, but it doesn’t work: “ValueError: Number of labels=1 does not match number of samples=4”.  It seems that the algorithm reads my ndarray as four univariate samples.

 

If I fit a ndarray of shape (n_samples, n_dims, n_timesteps), I get the following error: “ValueError: Found array with dim 3. Estimator expected <= 2.”

 

Can you help me?

 

Thanks again.

 

 

Da: Isak Samsten ***@***.***> 
Inviato: venerdì 26 novembre 2021 19:23
A: isaksamsten/wildboar ***@***.***>
Cc: LorenzoBonasera ***@***.***>; Author ***@***.***>
Oggetto: Re: [isaksamsten/wildboar] How is it possible to perform multivariate classification tasks? (Discussion #27)

 

Hi!

Many algorithms support multivariate time series out of the box (e.g., ShapeletForestClassifier). Just fit it with an ndarray of shape (n_samples, n_dims, n_timestep). Unfortunately,wildboar does not support datasets where each dim have different n_timestep. In the example folder there is an example with multivariate time series.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/isaksamsten/wildboar/discussions/27#discussioncomment-1707082> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWVQMVVCOLBNU6KK4NWGSOTUN7GAPANCNFSM5I27RHIA> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .   <https://github.com/notifications/beacon/AWVQMVSLVVWG7HENZDIG7U3UN7GAPA5CNFSM5I27RHIKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAANAYSQ.gif>

**Reply by isaksamsten** [reply]
Hi!

Multivariate support in the explain module has not been prioritised yet. The ShapeletForestCounterfactual explainer has support, but it has not been enable because I've not tested it enough. Create an issue and I'll try to allocate some time this week to enable support.
